roles/debian/nginx/tasks/main.yml

```diff
@@ -5,6 +5,14 @@
     state: present
     update_cache: true
     cache_valid_time: 240
+  register: nginx_installed
+
+- name: Remove default NGINX vhost that could clash with Varnish.
+  # We only need to do this once when Nginx is installed initially.
+  when: nginx_installed.changed
```

Comment:
Thanks for this! :smiley:

Two minor things though:
1. Because this might not always be desirable, can we couple this with a new default var to trigger the behaviour?
2. We have a running convention that variables created by registering or considered "private" begin with an underscore (think Drupal 5 :wink:) so could we please rename `nginx_installed` to `_nginx_installed`?

So defaults:

```yaml
nginx:
  keep_default_vhost: true
```
And your tasks:

```yaml
  register: _nginx_installed

- name: Remove default NGINX vhost that could clash with Varnish.
  when:
    - _nginx_installed.changed # We only need to do this once when Nginx is installed initially
    - not nginx.keep_default_vhost
  ansible.builtin.file:
    path: /etc/nginx/sites-enabled/default
    state: absent
```

That way we don't introduce any unexpected behaviour as the default behaviour is unchanged, and the underscore convention is respected.

Reply:
Oh, and the name, can we have this:

`Remove default NGINX vhost that could clash with other services running on port 80.`

Bit more generic. Sorry to be pedantic! :slightly_smiling_face:

Reply:
not sure we should add yet another nginx `keep_default_vhost` variable. The default behavior when `recreate_vhosts: true` is to wipe out all vhosts (also the default one). This would be very confusing when you have `recreate_vhosts: true` and `keep_default_vhost: true` because then the setting is ignored and all vhosts are wiped out anyway.

I think it is never desirable to keep the nginx default vhost directly after install, so I would always get rid of it.

Will address your other points.

Reply:
@gregharvey @tymofiisobchenko had the idea to check 2 conditions like this:
```yml
- name: Remove default NGINX vhost that could clash with other services running on port 80.
  when:
    # We only need to do this once when Nginx is installed initially.
    - _nginx_installed.changed
    - not nginx.recreate_vhosts
  ansible.builtin.file:
    path: /etc/nginx/sites-enabled/default
    state: absent
```

But this also does not make sense to me. @gregharvey wants to keep the default nginx config on load balancers for health checks. To prevent the default config from being wiped out you need to set `nginx.recreate_vhosts: false`. But exactly that setting would then match the condition here and the default config also gets wiped out.

No good ideas yet for an alternative proposal. Maybe we just keep our private workaround to install nginx first, then delete default config then call the debian/nginx role.

Reply:
Hold on. Do we need this at all? If we move these tasks to just after the package installation:
* https://github.com/codeenigma/ce-provision/blob/2.x/roles/debian/nginx/tasks/main.yml#L32-L38
* https://github.com/codeenigma/ce-provision/blob/2.x/roles/debian/nginx/tasks/main.yml#L92-L96

Then haven't we solved it? For Jobiqo you make sure `nginx.recreate_vhosts` is `true` and your default vhost *will* be removed. When that isn't desirable, set it to `false`. Problem solved, no extra task, just a re-jigging of existing task order. Or am I missing something?

Reply:
We need to set `nginx.recreate_vhosts` to false, so this does not run even if we move it around: https://github.com/codeenigma/ce-provision/blob/2.x/roles/debian/nginx/tasks/main.yml#L92-L96

The problem is that nginx ships a default hosts file. It is bad for servers with Varnish, so we need to remove it right after installation. This is what the changes in this pull request are doing.